### ISSUE TITLE
.prettierignore entries containing path separators are ignored

### ISSUE THREAD

**Issue by edmorley** [prettier/eslint-plugin-prettier#126]
**What version of `eslint` are you using?**
5.9.0

**What version of `prettier` are you using?**
1.15.2

**What version of `eslint-plugin-prettier` are you using?**
3.0.0

**Please paste any applicable config files that you're using (e.g. `.prettierrc` or `.eslintrc` files)**
https://github.com/edmorley/testcase-eslint-plugin-prettier-ignore/blob/master/.eslintrc.js
https://github.com/edmorley/testcase-eslint-plugin-prettier-ignore/blob/master/.prettierignore

**What source code are you linting?**
https://github.com/edmorley/testcase-eslint-plugin-prettier-ignore/blob/master/src/not_pretty.js

**Steps to reproduce**
1. Git clone https://github.com/edmorley/testcase-eslint-plugin-prettier-ignore
2. `yarn install`
3. `yarn prettier:check` # runs prettier
4. `yarn eslint:check` # runs eslint and so `eslint-plugin-prettier`

**What did you expect to happen?**
* No errors reported by either `yarn prettier:check` or `yarn eslint:check`, since the file with bad formatting is listed in `.prettierignore`.

**What actually happened?**
* No errors reported by `yarn prettier:check` (as expected)
* Formatting errors reported by `yarn eslint:check`:
```
$ yarn eslint:check
yarn run v1.12.3
$ eslint src

C:\Users\Ed\src\testcase-eslint-plugin-prettier-ignore\src\not_pretty.js
  1:12  error  Replace `⏎··'a'` with `"a"`  prettier/prettier

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

**Additional Notes**
* The error goes away if I edit the `.prettierignore` entry to not contain a path component (eg changing it to `not_pretty.js`)
* The variations `/src/not_pretty.js` and `./src/not_pretty.js` didn't help.
* This reproduced on both Windows and Ubuntu.

**Comment by BPScott** [comment]
Thanks for the report and the minimal reproduction repo (my favourite kind!)

It seems like Prettier's `getFileInfo()` only works out nested ignore paths correctly if you specify a relative filepath.

Replacing https://github.com/prettier/eslint-plugin-prettier/blob/master/eslint-plugin-prettier.js#L166-L168 with 

```
const relativeFilepath = require('path').relative(process.cwd(), filepath);
const prettierFileInfo = prettier.getFileInfo.sync(relativeFilepath, {
  ignorePath: '.prettierignore'
});
```

Seems to fix things, but I think this should be addressed upstream - `getFileInfo()` and `getFileInfo.sync()` should be able to work out ignores if you give it relative or absolute paths.

**Comment by BPScott** [comment]
Yo @edmorley, it looks like prettier 1.15.3 got released with a fix for this. It seems to work for me in your test repo. Can you give it a go and close this issue if it works for you.

**Comment by edmorley** [comment]
Hi! The testcase repository above now works as expected when prettier is upgraded to 1.15.3.

Testing against the original repository from which the testcase was derived didn't work initially, however this turned out to be due to `.eslintcache` not being invalidated after the prettier upgrade (that project passes `--cache` to eslint's CLI).

I've had a quick look to see if ESLint provides a way for plugins to contribute their own hash/version to the cache (which would help prevent this case), but couldn't find anything. Do you have any ideas? :-)

**Comment by edmorley** [comment]
> I've had a quick look to see if ESLint provides a way for plugins to contribute their own hash/version to the cache (which would help prevent this case), but couldn't find anything. Do you have any ideas? :-)

Hmm it appears that there isn't a good way of doing this at the moment:
https://github.com/eslint/eslint/issues/11060#issuecomment-436431695
